packages/frontend/src/features/sqlRunner/store/cartesianChartBaseSlice.ts

```diff
@@ -194,10 +195,33 @@ export const cartesianChartConfigSlice = createSlice({
             }
         },
         removeYAxisField: (state, action: PayloadAction<number>) => {
-            if (!state.config) return;
-            if (!state.config.fieldConfig) return;
+            if (!state.config?.fieldConfig?.y) return;
+
+            const index = action.payload;
+            state.config.fieldConfig.y.splice(index, 1);
+
+            if (state.config.display) {
+                if (state.config.display.yAxis) {
+                    state.config.display.yAxis.splice(index, 1);
+                }
 
-            state.config.fieldConfig.y.splice(action.payload, 1);
+                if (state.config.display.series) {
+                    Object.entries(state.config.display.series).forEach(
+                        ([key, series]) => {
+                            if (series.yAxisIndex === index) {
+                                // NOTE: Delete series from display object when it's removed from yAxis
+                                delete state.config?.display?.series?.[key];
+                            } else if (
+                                // NOTE: Decrease yAxisIndex for series that are after the removed yAxis
+                                series.yAxisIndex &&
+                                series.yAxisIndex > index
```

Comment:
Can you do this safely? 🤔

Reply:
yes we can because the index will always be > 0 with the current state 👍